lib/cmd/fh3/app/undeploy.js

```diff
@@ -1,51 +1,50 @@
 /* globals i18n */
-var log = require("../../../utils/log");
-var ini = require("../../../utils/ini");
-var util = require('util');
+
+/* globals i18n */
+var ini = require('../../../utils/ini');
+var fhc = require("../../../fhc");
 
 module.exports = {
-  'desc': i18n._('Undeploy an app given the domain and environment it is deployed to'),
-  'examples': [{
-    cmd: 'fhc app undeploy --id=<app-id> --env=<environment> --domain=<domain>',
-    desc: i18n._('Undeploy <app-id> from environment <environment> in domain <domain>')
-  }],
-  'demand': ['id', 'env'],
-  'alias': {
-    'id': 'i',
-    'env': 'e',
-    'domain': 'd',
-    0: 'id',
-    1: 'env',
-    2: 'domain'
-  },
-  'describe': {
-    'id': i18n._('Unique 24 character GUID of the application to undeploy'),
-    'env': i18n._('The ID of the environment from where to undeploy the app'),
-    'domain': i18n._('The name of the domain from where to undeploy the app'),
-    'embed': i18n._('If this flag is present then {embed: true} will be sent')
+  'desc': i18n._('Undeploy a application.'),
+  'examples' : [{ cmd : "fhc app undeploy --appid=<app-id> --apptype=<apptype> --env=<environment>", desc : i18n._('--apptype options:[cloud, embed] Default value is cloud')}],
+  'demand' : ['appid','env'],
+  'alias' : {
+    'appid' : 'a',
+    'apptype' : 'at',
+    'env' : 'e',
+    0 : 'app',
+    1 : 'apptype',
+    2 : 'env'
   },
-  'url': function (argv) {
-    return [
-      "api/v2/mbaas/deleteapp",
-      argv.domain || ini.get("domain"),
-      argv.env,
-      argv.id
-    ].join('/');
+  'describe' : {
+    'app_id' : i18n._('Unique 24 character GUID of the app you want to undeploy'),
+    'apptype' : i18n._('Options:[cloud, embed] - Default value is cloud'),
+    'env' : i18n._('Envinronment ID where the application is deployed')
   },
-  'method': 'post',
-  preCmd: function (params, cb) {
-    var request = {
-      embed: false,
-      id: params.id,
-      env: params.env,
-      domain: params.domain
-    };
+  'preCmd' : function(params, cb){
+    var type = params.apptype || "cloud";
 
-    if (typeof params.embed !== 'undefined') {
-      request.embed = true;
+    if (type !== "cloud" && !type !== "embed") {
+      return cb(i18n._("Invalid argument for --apptype: '"+type+"'"));
     }
 
-    log.info(util.format(i18n._("Embed flag set to %s"), request.embed));
-    return cb(null, request);
-  }
+    var deployTarget = ini.getEnvironment(params.env);
+
+    var dataObject = {
+      guid: fhc.appId(params.appid),
+      apptype: type,
+      deploytarget: deployTarget
+    };
+
+    return cb(undefined, dataObject);
+  },
+  postCmd : function(reponse, cb){
+    if (reponse.result == 'ok') {
+      return cb(null, i18n._("Undeploy completed successfully"));
+    }
+  },
+  'url' : function(argv){
+    return "box/srv/1.1/environments/" + argv.deploytarget + "/apps/" + argv.guid + "/undeploy";
```

Comment:
I don't think the old API is supposed to be used any more. Should use the v2 version (the same the one used in the studio.)